TestSupport/graph_generation_orchestrator.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+__all__ = []
+
+import multiprocessing
+import subprocess
+import argparse
+from pathlib import Path
+
+
+def call_geng(geng_path:str, order:int, num_edges:int, output_dir:Path):
+    """Call nauty geng as blocking process on multiprocessing thread
+
+    Opens a file for write (overwrites file if it exists) within the output_dir
+    and uses subprocess.run() to start a blocking process on the
+    multiprocessing pool thread to call the nauty geng executable with the
+    desired order and number of edges, with stdout redirected to the output
+    file object.
+
+    The resulting .g6 output file will contain all graphs of the desired order
+    for a single edge count.
+
+    Args:
+        geng_path: Path to the nauty geng executable
+        order: Desired number of vertices
+        num_edges: Desired number of edges
+        output_dir: Directory to which you wish to write the resulting .g6 file
+    """
+    filename = Path.joinpath(output_dir, f'n{order}.m{num_edges}.g6')
+    with open(filename, "w") as outfile:
+        subprocess.run(
+            [f'{geng_path}', f'{order}', f'{num_edges}:{num_edges}'],
+            stdout=outfile, stderr=subprocess.PIPE, shell=True)
```

Comment:
Happened to test this script on MacOS and found that the args weren't being passed as expected... However, when I allowed default `shell=False`, `subprocess.run()` passes the command line args as expected.

I'll have to test to see if default `shell=False` works on MacOS and Windows, and if not, then to use `shell=True` but to `[...].join()` the args into a string :(